server/api/studio/internal/service/datasource.go

```diff
@@ -282,21 +432,21 @@ func (d *datasourceService) testConnectionSFTP(host string, port int, username s
 	// connect to the remote SSH server
 	conn, err := ssh.Dial("tcp", fmt.Sprintf("%s:%d", host, port), config)
 	if err != nil {
-		return ecode.WithBadRequest(err, "failed to dial SSH server")
+		return ecode.WithErrorMessage(ecode.ErrBadRequest, err, "failed to dial SSH server")
 	}
 	defer conn.Close()
 
 	// create an SFTP client session
 	client, err := sftp.NewClient(conn)
 	if err != nil {
-		return ecode.WithBadRequest(err, "failed to create SFTP session")
+		return ecode.WithErrorMessage(ecode.ErrBadRequest, err, "failed to create SFTP session")
```

Comment:
ecode.WithBadRquest is the short-mode of ecode.WithErrorMessage(ecode.ErrBadRequest....

Reply:
![image](https://user-images.githubusercontent.com/18328704/226506145-d51dd736-a475-42a5-99bd-69f89489f350.png)
![image](https://user-images.githubusercontent.com/18328704/226506415-00e57791-69f8-4d6f-8202-e4e2ae23fb71.png)
its not same

Reply:
@veezhang why are they different? I previously think that the `WithBadRequest` is the shortcut of ht `WithErrorMessage(ecode.ErrBadRequest...`